src/qibo/models/dbi/double_bracket.py

```diff
@@ -171,24 +192,44 @@ def loss(self, step: float, d: np.array = None, look_ahead: int = 1):
         for _ in range(look_ahead):
             self.__call__(mode=self.mode, step=step, d=d)
 
-        # off_diagonal_norm's value after the steps
-        loss = self.off_diagonal_norm
+        # loss values depending on the cost function
+        if self.cost == DoubleBracketCostFunction.off_diagonal_norm:
+            loss = self.off_diagonal_norm
+        elif self.cost == DoubleBracketCostFunction.least_squares:
+            loss = self.least_squares(d)
+        elif self.cost == DoubleBracketCostFunction.energy_fluctuation:
+            loss = self.energy_fluctuation(self.state)
 
         # set back the initial configuration
         self.h = h_copy
 
         return loss
 
-    def energy_fluctuation(self, state):
+    def energy_fluctuation(self, state=None):
         """
         Evaluate energy fluctuation
 
         .. math::
-            \\Xi_{k}(\\mu) = \\sqrt{\\langle\\mu|\\hat{H}^2|\\mu\\rangle - \\langle\\mu|\\hat{H}|\\mu\\rangle^2} \\,
+            \\Xi(\\mu) = \\sqrt{\\langle\\mu|\\hat{H}^2|\\mu\\rangle - \\langle\\mu|\\hat{H}|\\mu\\rangle^2} \\,
 
         for a given state :math:`|\\mu\\rangle`.
 
         Args:
             state (np.ndarray): quantum state to be used to compute the energy fluctuation with H.
         """
-        return self.h.energy_fluctuation(state)
+        if state is None:
+            state = self.state
+        state_vector = np.zeros(len(self.h.matrix))
+        state_vector[state] = 1.0
+        return np.real(self.h.energy_fluctuation(state_vector))
+
+    def sigma(self, h: np.array):
+        return h - self.backend.cast(np.diag(np.diag(self.backend.to_numpy(h))))
+
+    def generate_Gamma_list(self, n: int, d: np.array):
+        r"""Computes the n-nested Gamma functions, where $\Gamma_k=[W,...,[W,[W,H]]...]$, where we take k nested commutators with $W = [D, H]$"""
+        W = self.commutator(d, self.sigma(self.h.matrix))
+        Gamma_list = [self.h.matrix]
```

Comment:
```suggestion
        gamma_list = [self.h.matrix]
```
Also for the other cases.